gusto/diagnostics.py

```diff
@@ -1717,7 +1723,36 @@ def setup(self, domain, state_fields):
             domain (:class:`Domain`): the model's domain object.
             state_fields (:class:`StateFields`): the model's field container.
         """
+
         m_X = state_fields(self.mixing_ratio_name)
         rho_d = state_fields(self.density_name)
+
+        m_X_space = m_X.function_space()
```

Comment:
I think we want all of this code to be protected behind an `if self.space_defined is None` -- currently this diagnostic allows the user to pass in a `space` (and we want the user to be able to do that) but it will be overridden by this code. I think we only want to override if `space is None`

Reply:
Cool, I've changed it to enable this, although I've used `if self.space is None` instead of `self.space_defined`